presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformExistsApplyToCorrelatedJoin.java

```diff
@@ -108,6 +108,19 @@ public Result apply(ApplyNode parent, Captures captures, Context context)
             return Result.empty();
         }
 
+        /*
+        Empty correlation list indicates that the subquery contains no correlation symbols from the
+        immediate outer scope. The subquery might be either not correlated at all, or correlated with
+        symbols from further outer scope.
+        Currently, the two cases are indistinguishable.
+        To support the latter case, the ApplyNode with empty correlation list is rewritten to default
+        aggregation, which is inefficient in the rare case of uncorrelated EXISTS subquery,
+        but allows to successfully decorrelate a correlated EXISTS subquery.
+         */
+        if (parent.getCorrelation().isEmpty()) {
+            return Result.ofPlanNode(rewriteToDefaultAggregation(parent, context));
```

Comment:
what this `if` changes? What `rewriteToNonDefaultAggregation` would return when `parent.getCorrelation().isEmpty()`?

Reply:
In the case of empty correlation, `rewriteToNonDefaultAggregation` would fail (return `Optional.empty()`), and eventually `rewriteToDefaultAggregation` would be triggered.
This was caused by certain behavior of the `PlanNodeDecorrelator` class.

Adding this shortcut doesn't change the resulting plan, but removes the responsibility from `PlanNodeDecorrelator`.